plugins/chan_track.py

```diff
@@ -727,7 +727,8 @@ def _parse_mode_string(modes, params, status_modes, mode_types):
                 mode_type = 'B' if is_status else None
 
             if mode_type in "AB" or (mode_type == 'C' and adding):
-                param = params.pop(0)
+                if len(params) != 0:
+                    param = params.pop(0)
```

Comment:
There is no case where a parameter of type `A`, `B`, or `C` (if adding) would not have a parameter, what error is this intending to fix specifically?

Reply:
[00:00:39] [INFO] [snoonet|out] >> b'JOIN #nekotest\r\n'
[00:00:39] [ERROR] Error in hook chan_track:on_mode
Traceback (most recent call last):
  File "C:\Users\katsw\PycharmProjects\CloudBot\cloudbot\plugin.py", line 456, in internal_launch
    out = yield from task
  File "D:\Python\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\katsw\PycharmProjects\CloudBot\cloudbot\plugin.py", line 423, in _execute_hook_threaded
    return call_with_args(hook.function, event)
  File "C:\Users\katsw\PycharmProjects\CloudBot\cloudbot\util\func_utils.py", line 19, in call_with_args
    return func(*args)
  File "C:\Users\katsw\PycharmProjects\CloudBot\plugins\chan_track.py", line 762, in on_mode
    new_modes = _parse_mode_string(modes, mode_params, status_modes, mode_types)
  File "C:\Users\katsw\PycharmProjects\CloudBot\plugins\chan_track.py", line 730, in _parse_mode_string
    param = params.pop(0)
IndexError: pop from empty list

Reply:
NekoBot -x
NekoBot +SW